src/reducers/logDialogsReducer.ts

```diff
@@ -7,41 +7,77 @@ import { AT } from '../types/ActionTypes'
 import { replace } from '../Utils/util'
 import produce from 'immer'
 
-const initialState: LogDialogState = [];
+const initialState: LogDialogState = {
+    logDialogs: [],
+    continuationToken: undefined,
+    pendingDelete: []
+}
 
 const logDialogsReducer = produce((state: LogDialogState, action: ActionObject) => {
     switch (action.type) {
         case AT.USER_LOGOUT:
-            return [...initialState]
+            return { ...initialState }
         case AT.FETCH_LOG_DIALOGS_FULFILLED:
-            return action.allLogDialogs
+            if (action.clear) {
+                state.logDialogs = action.logDialogs
+                state.continuationToken = action.continuationToken
+            }
+            else {
+                state.logDialogs = [...state.logDialogs, ...action.logDialogs]
+                state.continuationToken = action.continuationToken
+            }
+            return
         case AT.FETCH_LOG_DIALOG_FULFILLED:
             if (action.replaceLocal) {
-                const existingLogDialog = state.find(d => d.logDialogId === action.logDialog.logDialogId)
+                const existingLogDialog = state.logDialogs.find(d => d.logDialogId === action.logDialog.logDialogId)
                 if (existingLogDialog) {
-                    return replace(state, action.logDialog, a => a.logDialogId)
+                    replace(state.logDialogs, action.logDialog, a => a.logDialogId)
                 }
                 else {
-                    return [...state, action.logDialog]
+                    state.logDialogs = [...state.logDialogs, action.logDialog]
```

Comment:
Given this list is likely large using push here is more appropriate. `state.logDialogs.push(action.logDialog)`